code/mission/missioncampaign.h

```diff
@@ -124,13 +125,15 @@ class campaign
 	ubyte	ships_allowed[MAX_SHIP_CLASSES];		// which ships the player can use
 	ubyte	weapons_allowed[MAX_WEAPON_TYPES];		// which weapons the player can use
 	cmission	missions[MAX_CAMPAIGN_MISSIONS];	// decription of the missions
-	int				num_variables;					// number of variables this campaign had - Goober5000
-	sexp_variable	*variables;						// malloced array of sexp_variables (of num_variables size) containing campaign-persistent variables - Goober5000
-	int				redalert_num_variables;			// These two variables hold the previous state of the above for restoration
-	sexp_variable	*redalert_variables;			// if you replay the previous mission in a Red Alert scenario. -MageKing17
+	//int				num_variables;					// number of variables this campaign had - Goober5000
+	//sexp_variable	*variables;						// malloced array of sexp_variables (of num_variables size) containing variables which are saved - Goober5000
+	//int				redalert_num_variables;			// These two variables hold the previous state of the above for restoration
+	//sexp_variable	*redalert_variables;			// if you replay the previous mission in a Red Alert scenario. -MageKing17
+	SCP_vector<sexp_variable> persistent_variables;
+	SCP_vector<sexp_variable> red_alert_data;
```

Comment:
`red_alert_data` isn't a good name for this since it does not mention that this is variable data. What was wrong with the old `redalert_variables` name?

Also, this could use a comment to explain what it is used for.